doc/usage/configuration.rst

```diff
@@ -2945,6 +2969,18 @@ Options for the Python domain
 
    .. note:: This configuration is still in experimental
 
+.. confval:: python_maximum_signature_line_length
+
+   An integer representing the maximum number of characters that cannot be exceeded
+   by a Python object's signature. When negative (the default), there is no maximum,
+   no line break will be introduced no matter how long the signature. When positive,
+   all objects whose signature exceed the given character limit will have each of
+   their arguments displayed on a separate, indented line. The directive
+   :rst:dir:`single-line-signature` allows to disable this behavior on specific
+   objects.
+
+   .. versionadded:: 6.x
```

Comment:
I've mixed feelings on this, how would you feel about just always breaking to one line per argument and having a global (boolean) option to preserve single lines?

A

Reply:
As a potential user of this, I'd really like to be able to specify a threshold (and have Sphinx ship with a sensible default...)
I don't want simple functions with one or two args having their signatures split across lines, and I have an even enough mix that a boolean is going to cause an annoying amount of work no matter which way it is set....

Reply:
I tend to agree with @cjw296 -- something like https://installer.pypa.io/en/stable/api/__init__/ should stay on all-entries-on-one-line (instead of breaking that up) because they fit, while something like https://testfixtures.readthedocs.io/en/latest/api.html#testfixtures.compare should get split across multiple lines.

Using the number of characters in the signature is the natural piece for enable that; and the explicit opt-outs on a per-project and per-signature level are also useful to get the exact semantics that an author might want.

As a user, I'd be... annoyed÷frustrated with the proposal that @AA-Turner has made; *especially* if the `installer` API example would get broken up into multiple lines after a Sphinx minor/patch update. As a theme author, I'd implore you to, should you decide to change the default behaviour like that, to handle it like a backwards-incompatible change to a public Python API of this project. :)

Reply:
re "When negative (the default), there is no maximum, no line break will be introduced no matter how long the signature".
I assume that **if** the html theme sets a maximum body width (either through a configuration parameter or hardcoded), then the negative default value will not override this, i.e. the signature will be wrapped based on html theme. Perhaps that could be noted/documented in the description of the `python_maximum_signature_line_length` parameter.
For example, see `basic` html theme's `body_max_width` [parameter](https://www.sphinx-doc.org/en/master/usage/theming.html).

Reply:
@emezh yes indeed, you can see the line breaks introduced by `python_maximum_signature_line_length` as hard-wrapping, while the soft-wrapping in the view (here the HTML render) can wrap lines independently. I did not make this clear at all in the doc though, so I clarified this point in my last commit ([0bc3b68](https://github.com/sphinx-doc/sphinx/pull/11011/commits/0bc3b68d34c4d49f7cc71413a6d83dd74ef12f2d))